bitnami/mongodb/templates/common-scripts-cm.yaml

```diff
@@ -101,4 +103,19 @@ data:
     shopt -s extglob
     rm -rf !(mongodb-ca-cert|mongodb.pem|CAs|openssl.cnf)
     chmod 0600 mongodb-ca-cert mongodb.pem
+    {{- else }}
+    {{- if eq .Values.architecture "standalone" }}
+      ID="0"
+    {{- else }}
+    if [[ "$MY_POD_NAME" =~ "arbiter-0"$ ]]; then
+      ID="0"
+    elif [[ "$MY_POD_NAME" =~ "hidden-"[0-9]{1,}$ ]]; then
+      ID="${MY_POD_NAME#"{{ printf "%s-hidden-" $fullname }}"}"
+    else
+      ID="${MY_POD_NAME#"{{ $fullname }}-"}"
+    fi
+    {{- end }}
+    cat "/certs-${ID}/tls.crt" "/certs-${ID}/tls.key" > "/certs/mongodb.pem"
+    cp "/certs-${ID}/ca.crt" "/certs/mongodb-ca-cert"
```

Comment:
I think you also need to configure the right permissions for the certificates here, as with the generated ones:

```diff
    cat "/certs-${ID}/tls.crt" "/certs-${ID}/tls.key" > "/certs/mongodb.pem"
    cp "/certs-${ID}/ca.crt" "/certs/mongodb-ca-cert"
+   chmod 0600 /certs/mongodb-ca-cert /certs/mongodb.pem
```

The individual provided files may be mounted with the right permissions, but the generated ones (concatenating using `cat` or copying with `cp`) will more than likely not.

Reply:
Agree! Changed